### PR TITLE
Feature: kids 38 add more amplitude tracking

### PR DESCRIPTION
**Pull request by Daniela510**
## Description

- Remove vague 'button pressed' event & replace with explicit events;
- Add wallet tracker of all kids when a new user is set
- Differentiate the 2 'coin flows' by adding deeplink events and in app scanning events
- Added other miscellaneous events we would want to track.

### CODE REVIEW

File: lib/helpers/analytics_helper.dart [hunk 1]
```diff
@@ -56,12 +55,21 @@ class AnalyticsHelper {
     await _amplitude!.trackingSessionEvents(true);
   }
 
-  static Future<void> setUserId(String profileName) async {
+  static Future<void> setUserId(
+      String profileName, List<Profile> profiles) async {
     final currentUserId = await _amplitude?.getUserId();
     final isNewUser = profileName != currentUserId;
+    Map<String, double> walletTracker = {
```
Comment (MaikelStuivenberg) [inline lib/helpers/analytics_helper.dart:62]: I wouldn't put this here.
This information is available (in this way) in Retool and that one is always up to date. In Amplitude its really depending on if someone is opening the app.

**Review by MaikelStuivenberg — CHANGES_REQUESTED** [review]

File: lib/features/profiles/screens/wallet_screen.dart [hunk 2]
```diff
@@ -130,14 +130,21 @@ class _WalletScreenState extends State<WalletScreen>
...
-                            context: context,
-                            shape: RoundedRectangleBorder(
-                              borderRadius: BorderRadius.circular(10.0),
-                            ),
-                            builder: (context) => const GiveBottomSheet(),
-                          ),
-                        )
+                          onPressed: () {
+                            AnalyticsHelper.logEvent(
+                                eventName: AmplitudeEvent.iWantToGiveToPressed,
```
Comment (pavelpavelko) [inline lib/features/profiles/screens/wallet_screen.dart:135]: perhaps iWantToGiveToPressed -> iWantToGivePressed?

File: lib/features/scan_nfc/cubit/scan_nfc_cubit.dart [hunk 3]
```diff
@@ -77,6 +81,9 @@ class ScanNfcCubit extends Cubit<ScanNfcState> {
           scanNFCStatus: ScanNFCStatus.error,
           coinAnimationStatus: CoinAnimationStatus.stopped));
       NfcManager.instance.stopSession();
+      AnalyticsHelper.logEvent(
```
Comment (pavelpavelko) [inline lib/features/scan_nfc/cubit/scan_nfc_cubit.dart:84]: shall we also track mediumID when coin is scanned?

File: lib/helpers/analytics_helper.dart [hunk 4]
```diff
@@ -35,17 +42,9 @@ enum AmplitudeEvent {
...
-  // static const String newTransactionKey = "new_transaction";
-  // static const String amountKey = "amount";
-  // static const String goalKey = "goal_name";
-  // static const String walletAmountKey = "wallet_amount";
-  // static const String buttonPressedKey = "button_pressed";
-  // static const String buttonNameKey = "button_name";
-  // static const String screenNameKey = "screen_name";
-  // static const String timestampKey = "timestamp";
-  // static const String formattedDateKey = "formatted_date";
+  static const String amountKey = "amount";
```
Comment (pavelpavelko) [inline lib/helpers/analytics_helper.dart:45]: I like it)

**Review by pavelpavelko — APPROVED** [review]

File: lib/helpers/analytics_helper.dart (hunk 1 above)
Comment (Daniela510) [reply lib/helpers/analytics_helper.dart:62]: A wallet tracker was part of the story's acceptance criteria. Not sure where that request came from, but it's no problem to delete

File: lib/features/scan_nfc/cubit/scan_nfc_cubit.dart (hunk 3 above)
Comment (Daniela510) [reply lib/features/scan_nfc/cubit/scan_nfc_cubit.dart:84]: Yes, I added it but in this case its an error so nothing was scanned from the NFC

**Review by MaikelStuivenberg — APPROVED** [review]

**Pull request merged**